### PR TITLE
[mmp] Set ThrowManagedException in debug apps if unset

### PR DESCRIPTION
**Pull request by chamons**
- https://github.com/xamarin/xamarin-macios/issues/5738
- There are a number of managed exceptions Apple can throw at you during
debugging, such as expanding a NSColor in the wrong colorspace
- Throwing a managed exception is a nicer debugging experience, and
during debug we don't care about any performance penality.

### CODE REVIEW

File: tools/mmp/driver.cs [hunk 1]
```diff
@@ -379,6 +379,11 @@ static void Main2 (string [] args)
 			if (!targetFramework.HasValue)
 				targetFramework = TargetFramework.Default;
 
+			if (App.EnableDebug && App.MarshalObjectiveCExceptions == MarshalObjectiveCExceptionMode.Default) {
+				Log (1, $"Selecting ThrowManagedException for marshal-objectivec-exceptions as Application is in debug");
```
Comment (wachaudh) [inline tools/mmp/driver.cs]: string interpolation here seems necessary

**Comment by chamons** [comment]
Broke a legit test (changed assumptions):
```
 MonoTouchFixtures.ObjCRuntime.ExceptionsTest.ManagedExceptionPassthrough: 1 objc mode
Expected: UnwindManagedCode
But was: ThrowManagedException

MonoTouchFixtures.ObjCRuntime.ExceptionsTest.ObjCException: objc mode
Expected: UnwindManagedCode
But was: ThrowManagedException
```

**Comment by mandel-macaque** [comment]
Approving assuming tests are fixed, else do not land ;)

File: tools/mmp/driver.cs [hunk 2]
```diff
@@ -379,6 +379,11 @@ static void Main2 (string [] args)
 			if (!targetFramework.HasValue)
 				targetFramework = TargetFramework.Default;
 
+			if (App.EnableDebug && App.MarshalObjectiveCExceptions == MarshalObjectiveCExceptionMode.Default) {
```
Comment (rolfbjarne) [inline tools/mmp/driver.cs]: This is the right place to set the default value: https://github.com/xamarin/xamarin-macios/blob/cd9a151c984d814fb4f5ac83291620fc206d4ec7/tools/common/Application.cs#L362-L377

Comment (chamons) [reply tools/mmp/driver.cs]: :man_facepalming:

**Comment by mandel-macaque** [comment]
Known issue: https://github.com/xamarin/maccore/issues/581

**Pull request merged**